### PR TITLE
Upgrade eslint peer deps

### PR DESCRIPTION
**Pull request by emmenko**
There are some major eslint version bumps.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -13,6 +13,7 @@ module.exports = {
 
   env: {
     es6: true,
+    jest: true,
```
Comment (Siilwyn) [inline index.js]: Assume you are using this for a new project?
No way to specify this in the project itself?

Comment (emmenko) [inline index.js]: Yeah, this is what we are doing at the moment. The idea is to move to `jest` in general anyway.
But it doesn't matter much if some flags are defined here (e.g. `mocha` is also defined here, although it's not used in every project).

I don't have a strong opinion on this, I'm fine with what we decide.

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-config-commercetools",
-  "version": "4.0.2",
+  "version": "5.0.0",
```
Comment (emmenko) [inline package.json]: Because of the major eslint version bumps, we should bump a major version as well

**Comment by emmenko** [comment]
Guys, happy with that?

**Comment by hisabimbola** [comment]
Just a quick thought

For the user to update this package to the new version?
They need to update the existing peerDependencies saved in their package.json

Any need to add that to the README?

**Comment by emmenko** [comment]
> For the user to update this package to the new version?
They need to update the existing peerDependencies saved in their package.json

It should be fine like this. If they do upgrade to the new version, npm will complain that the other dependencies do not fit the range define in the peerDeps.
This is standard behaviour, there is no to mention this.

**Comment by Siilwyn** [comment]
👍

**Comment by emmenko** [comment]
🚀

**Pull request merged**